examples/esp32/xively_demo/main/main.c

```diff
@@ -0,0 +1,293 @@
+/* Copyright (c) 2003-2016, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client codebase,
+ * it is licensed under the BSD 3-Clause license.
+ */
+#include <string.h>
+#include <errno.h>
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/event_groups.h"
+#include "esp_event_loop.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "esp_log.h"
+#include "esp_err.h"
+#include "esp_vfs_fat.h"
+
+#include "xively_if.h"
+#include "gpio_if.h"
+#include "provisioning.h"
+#include "user_data.h"
+
+#ifndef USE_HARDCODED_CREDENTIALS
+/* Default workflow uses runtime provisioning and flash storage for user data */
+  #define USE_HARDCODED_CREDENTIALS 0
+#endif /* USE_HARDCODED_CREDENTIALS */
+
+#if USE_HARDCODED_CREDENTIALS
+  #define USER_CONFIG_WIFI_SSID     "[SET YOUR WIFI NETWORK NAME HERE]"
+  #define USER_CONFIG_WIFI_PWD      "[SET YOUR WIFI NETWORK PASSWORD HERE]"
+  #define USER_CONFIG_XI_ACCOUNT_ID "[SET YOUR XIVELY ACCOUNT ID HERE]"
+  #define USER_CONFIG_XI_DEVICE_ID  "[SET YOUR XIVELY DEVICE  ID HERE]"
+  #define USER_CONFIG_XI_DEVICE_PWD "[SET YOUR XIVELY DEVICE PASSWORD HERE]"
+#endif /* USE_HARDCODED_CREDENTIALS */
+
+#define XIF_TASK_ESP_CORE    0 /* ESP32 core the XIF task will be pinned to */
+#define XIF_TASK_STACK_SIZE  36 * 1024
+#define GPIO_TASK_STACK_SIZE 2 * 1024
+
+#define WIFI_CONNECTED_FLAG  BIT0
+
+static EventGroupHandle_t app_wifi_event_group;
+static user_data_t user_config;
+
+static esp_err_t app_wifi_event_handler( void* ctx, system_event_t* event )
+{
+    //printf( "\nNew WiFi event: ID [%d]", event->event_id );
```

Comment:
remove?

Reply:
This function is key for the "crash on AP disconnection" issue, so I'd rather leave these comments in for now. I'll make sure to uncomment them or remove them when I'm done with the crash issue. Not uncommenting them now because I'm not sure whether they have any effect on the crash.